src/test/java/org/osiam/client/BugFilterWithNot.java

```diff
@@ -0,0 +1,109 @@
+package org.osiam.client;
+
+import static org.hamcrest.CoreMatchers.hasItem;
+import static org.hamcrest.CoreMatchers.not;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.fail;
+
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.osiam.client.oauth.AccessToken;
+import org.osiam.client.oauth.Scope;
+import org.osiam.client.query.Query;
+import org.osiam.client.query.QueryBuilder;
+import org.osiam.resources.scim.Group;
+import org.osiam.resources.scim.MemberRef;
+import org.osiam.resources.scim.SCIMSearchResult;
+import org.osiam.resources.scim.User;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.TestExecutionListeners;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import org.springframework.test.context.support.DependencyInjectionTestExecutionListener;
+
+import com.github.springtestdbunit.DbUnitTestExecutionListener;
+import com.github.springtestdbunit.annotation.DatabaseOperation;
+import com.github.springtestdbunit.annotation.DatabaseSetup;
+import com.github.springtestdbunit.annotation.DatabaseTearDown;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@ContextConfiguration("/context.xml")
+@TestExecutionListeners({ DependencyInjectionTestExecutionListener.class,
+        DbUnitTestExecutionListener.class })
+@DatabaseSetup("/database_seeds/SearchUserServiceIT/database_seed_bug_filter_with_not.xml")
+@DatabaseTearDown(value = "/database_tear_down.xml", type = DatabaseOperation.DELETE_ALL)
+public class BugFilterWithNot extends AbstractIntegrationTestBase {
+    private static final String ADMIN_USER_NAME = "marissa";
+    private static final String ADMIN_USER_PASSWORD = "koala";
+    private static final String NAME_GROUP_TWO = "test_group02";
+    private final String TEST_USER_NAME = "marissa";
+    private final String TEST_USER_ID = "cef9452e-00a9-4cec-a086-d171374ffbef";
+
+    private AccessToken accessToken;
+
+    @Before
+    public void setup() {
+        accessToken = OSIAM_CONNECTOR.retrieveAccessToken(ADMIN_USER_NAME, ADMIN_USER_PASSWORD, Scope.ADMIN);
+    }
+
+    @Ignore("This test fails at the moment because the bug is not yet fixed. @Ignore must be removed after the bug-fix.")
+    @Test
+    public void filter_groups_where_the_user_is_not_included_in() throws Exception {
+        // check precondition
+        Group expectedGroup = getGroupByName(NAME_GROUP_TWO);
+        assertNotContainsUser(expectedGroup, TEST_USER_ID);
```

Comment:
I don't think this assertion is necessary, after it is just asserting that the test data as described in the seed is actually in the DB. I would drop it.